### ISSUE TITLE
Remove cy.server() boilerplate

### ISSUE THREAD

**Issue by paulfalgout** [cypress-io/cypress#439]
There's probably a really good reason not to do this, but I thought I'd float it just in case.

If you call `cy.route()` without first calling `cy.server()` you get `CypressError: cy.route() cannot be invoked before starting the cy.server()`

Totally makes sense.. but in every Cypress test I've written thus far my only use case for `cy.server()` is adding it prior to the first `cy.route()` call.

Is there any reason why, instead of erroring it could just call `cy.server()` for me?

**Comment by brian-mann** [comment]
I agree. The idea behind `cy.server` was that it would provide all of the defaults for all of the subsequent `cy.route` calls.

But I imagine most if not all users don't pass anything into it. In that sense it just becomes "that one thing you do before cy.route".

There is a looming network traffic proxy rewrite coming soon, and we were going to remove it then, because we will add breaking changes to the routing API anyway.

**Comment by jennifer-shehane** [comment]
The features requested in this issue are now possible as part of `cy.route2()`. 

`cy.route2()` is currently experimental and requires being enabled by passing `"experimentalNetworkStubbing": true` through your Cypress configuration. This will eventually be merged in as part of our standard API. 

**Please see the `cy.route2()` docs for full details: https://on.cypress.io/route2**

If you encounter any issues or unexpected behavior while using `cy.route2()` we encourage you to [open a new issue](https://github.com/cypress-io/cypress/issues/new?assignees=&labels=&template=bug-report.md&title=) so that we can work out all the issues before public release. Thanks!
